src/classes/ACCT_IndividualAccounts_TDTM.cls

```diff
@@ -873,54 +795,184 @@ public class ACCT_IndividualAccounts_TDTM extends TDTM_Runnable {
+                }
+            }
+        }
+
+        if (listOpp.size() > 0) {
+            UTIL_DMLService.updateRecords(listOpp);
+        }
+    }
+
+    /*******************************************************************************************************
+    * @description for the list of HH AccountId's, update their Household Names
+    * @param accountIds the list of Account (Id's) that need updating.
+    * @return void
+    */
+    public static void renameHHAccounts(List<Id> accountIds) {
+        if (accountIds.isEmpty()) {
+            return;
+        }
+
+        if (accountIds.size() == 1 || system.isFuture() || system.isBatch()) {
+            HH_HouseholdNaming hhName = new HH_HouseholdNaming();
+            hhName.UpdateNames(accountIds);
+        } else {
+            HH_HouseholdNaming.FutureUpdateNames(accountIds);
+        }
+    }
+
+    /*******************************************************************************************************
+    * @description utility to return the recordtypeId for a new account, given the account model.
+    * @return Id. if none is explicitly specified, returns null.
+    */
+    public static Id rtIdForNewAccount() {
+        return rtIdForAccountModel(CAO_Constants.isHHAccountModel());
+    }
+
+    /*******************************************************************************************************
+    * @description utility to return the recordtypeId for a new account, given the account model.
+    * @param isHHAccountModel whether we are using the Household Account Model
+    * @return Id. recordtype Id to use, or null if none is specified for the account model.
+    */
+    public static Id rtIdForAccountModel(Boolean isHHAccountModel) {
+        contactsSettings = UTIL_CustomSettingsFacade.getContactsSettings();
+        Schema.Recordtypeinfo rtById = null;
+        // check if recordtypes are enabled, and if it matches the existing setting
+        // we go thru describe to validate that it is a valid recordType ID.
+        Id rtId = isHHAccountModel ? contactsSettings.npe01__HH_Account_RecordTypeID__c : contactsSettings.npe01__One_to_One_RecordTypeID__c;
+        if (rtId != null) {
+            Schema.DescribeSObjectResult R = Account.SObjectType.getDescribe();
+            Map<Id,Schema.RecordTypeInfo> rtMapById = r.getRecordTypeInfosById();
+            rtById = rtMapById.get(rtId);
+            return (rtById == null ? null : rtById.getRecordTypeID());
+        }
+        return null;
+    }
+
+    /*******************************************************************************************************
+    * @description for the list of AccountId's, recalc their rollups 
+    * @param accountIds the list of Account (Id's) that need updating.
+    * @return void
+    */ 
+    public static void rollupAccounts(List<Id> accountIds) {
+        Boolean isImmediate = accountIds.size() == 1 || System.isFuture() || System.isBatch() || System.isQueueable();
```

Comment:
Since a Queueable can queue up another Queueable, does it need to force immediate if being executed within a queueable?

Reply:
@force2b - no but I believe the idea is to push it off into an async method so if we are in an async method there would be no need to hold off.